src/scripts/sync_subclassof.py

```diff
@@ -0,0 +1,589 @@
+    pass  # no output for this case
+
+    # Case 3: SCR is direct in the source, but not at all in Mondo
+    # todo: indirect+direct should == ancestors. Why not?
+    #   1. code
+    #    ancestors_mondo2 = rels_indirect_mondo_mondo.union(rels_direct_mondo_mondo)
+    #    missing_ancestors = ancestors_mondo2.difference(ancestors_mondo)  # n=~500
+    #   2. use ancestors_mondo? - _edges_with_metadata_from_source_edges(ancestors_mondo_source)
+    #   3. should it be called ancestors_mondo_mondo? since they're mondo ids from mondo
+    # - normalize IDs to same namespace
+    # todo: better to normalize to source IDs than mondo?
+    #  since ids may be missing in mondo? but we want to drop those anyway right?
+    ancestors_mondo2_mondo = rels_indirect_mondo_mondo.union(rels_direct_mondo_mondo)
+    ancestors_mondo2_source, ancestors_mondo2_and_1or2_ids_not_in_source = _convert_edge_namespace(
+        ancestors_mondo2_mondo, mondo_source_map)
+    # - set difference
+    in_source_direct_not_in_mondo_source_edges = rels_direct_source.difference(ancestors_mondo2_source)
+    in_source_direct_not_in_mondo: List[Dict] = _edges_with_metadata_from_plain_edges(
+        in_source_direct_not_in_mondo_source_edges, source_data_map)
+    # - create dataframe
+    # df3 = pd.DataFrame(in_source_only_direct)  # attempt 1: was wrong: was only counting mondo direct, not all rels
+    df3 = pd.DataFrame(in_source_direct_not_in_mondo)
+    if len(df3) == 0:
+        df3 = pd.DataFrame(columns=list(ROBOT_SUBHEADER[0].keys()))
+    # - warning cases
+    #   only object_mondo_id should ever be missing, but checking both to be safe
+    unmigrated_terms = set(df3[df3['object_mondo_id'].isna()]['object_source_id']).union(
+        set(df3[df3['subject_mondo_id'].isna()]['object_source_id']))
+    if len(unmigrated_terms) >= 200:
+        logging.warning(f'Warning: {len(unmigrated_terms)} unmigrated source terms found in "{ontology_name}"; '
+                        f'higher number than expected.')
+    # - format
+    #   only object_mondo_id should ever be missing, but checking both to be safe
+    df3 = df3[~df3['subject_mondo_id'].isna() & ~df3['object_mondo_id'].isna()].sort_values(['subject_mondo_id'])
+    # - obsolete cases
+    robot_subheader_obs = deepcopy(ROBOT_SUBHEADER)
+    robot_subheader_obs[0]['object_mondo_id'] = 'MONDO:excluded_subClassOf'
+    df3_obs = df3[df3['subject_mondo_label'].str.startswith('obsolete') |
+       df3['object_mondo_label'].str.startswith('obsolete')]
+    # - non-obsolete cases
+    df3 = df3[~(df3['subject_mondo_label'].str.startswith('obsolete') |
+        df3['object_mondo_label'].str.startswith('obsolete'))]
+    # - save
+    df3_obs = pd.concat([pd.DataFrame(robot_subheader_obs), df3_obs])
+    df3_obs.to_csv(outpath_added_obsolete, sep='\t', index=False)
+    df3 = pd.concat([pd.DataFrame(ROBOT_SUBHEADER), df3])
+    df3.to_csv(outpath_added, sep='\t', index=False)
```

Comment:
## Bug: Case 5: Getting diffs when rerunning (@joeflack4)
For some reason, I rerun and every time I do so, I'm getting a diff showing up in `git status`. That should not be the case. I should improve sorting, and then after that if there is still a diff after running 2x in a row, I need to investigate further.

Reply:
@matentzn You were correct that sorting was the problem. That was my first go-to too. I had forgotten already what I recently learned in #368, that 1 Mondo ID -> 2+ source IDs is allowed, and is called proxy merging. I was counting on that not being the case, so I wasn't sorting on enough columns. Every Friday I go over a list of things I'm trying to memorize; I'll add this fact to that list so that I can try to ensure I remember this detail.